integration_test.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+set -Eemo pipefail
+# This script is meant to be run inside https://hub.docker.com/_/docker
+# See .github/workflows/ci.yml:18 for how to run this yourself
+
+# Wait for the docker daemon to start up
+for i in {1..30}; do
+    if docker ps; then
+        break
+    else
+        sleep 1
+    fi
+done
+if [ "$i" = '30' ]; then
+    echo "Docker did not start within 30 seconds, aborting"
+    exit 1
+fi
+
+# Start docuum in the background
+/artifacts/docuum-x86_64-unknown-linux-musl --threshold=13MB &
+sleep 1
+# Wait for docuum to start sleeping by checking the process state
+function wait {
+    # http://man7.org/linux/man-pages/man5/proc.5.html
+    DOCUUM_PID=$(pgrep docuum-x86_64-unknown-linux-musl)
+    for i in {1..60}; do
+        if [ "$(awk '{ print $3 }' /proc/${DOCUUM_PID}/stat)" = 'S' ]; then
+            break
+        fi
+        echo "Waiting for docuum to finish one loop..."
+        sleep 1
+    done
+    if [ "$i" = '60' ]; then
+        echo "Docuum did not finish a loop within 1 minute, aborting"
+        exit 1
+    fi
```

Comment:
I wonder if it would make sense to abstract away this "wait for something to succeed" logic into a reusable function, and then use it to do both of these wait loops?

Alternatively, one possibly bad idea is to start Docker and Docuum at the same time (Docuum should be fine starting before Docker is ready, I think), and then wait for both Docker and Docuum in the same loop? Or you could even just wait for Docuum (and not Docker), because Docuum itself will keep retrying until Docker is ready.

Reply:
I tested out removing the "waiting for docker to startup" code, but what happens is that Docuum sleeps for 1 second when it can't connect, which tricks this script into thinking docuum has finished it's first loop and it has gone back to sleep. So we need to wait for docker to start up separately.

And about abstracting into a method: I like to follow the "rule of 3": only refactor something into a function if it's used 3 or more times. Especially since I'd have to find a way to parameterize the wait conditions (`docker ps` in one and `awk` in the other)